apps/user-office-backend/src/eventHandlers/customHandler.ts

```diff
@@ -215,12 +215,10 @@ export default function createCustomHandler() {
             });
 
           if (allProposalsOnCall && allProposalsOnCall.proposalViews.length) {
-            await Promise.all(
-              allProposalsOnCall.proposalViews.map(
-                async (proposalOnCall) =>
-                  await handleWorkflowEngineChange(event, proposalOnCall)
-              )
+            const proposalPks = allProposalsOnCall.proposalViews.map(
+              (proposal) => proposal.primaryKey
             );
+            handleWorkflowEngineChange(proposalDataSource, event, proposalPks);
```

Comment:
Why are we sending the reference of datasource to the handler, can it not resolve this dependency itself or is there some circular dependency?

Reply:
@enthusiastio I wasn't sure if this is a good idea and that's why I decided to send it from outside. Because this handler is called quite frequently I was thinking instead of resolving that on every single call I can just send it from outside and resolve only once when the handler is created. Not sure if this makes any difference or not but I guess that was the main reason.

Reply:
I think it doesn't make any difference and I will try to resolve it in the workflow engine change handler instead. Thanks for the good comments.